components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -866,6 +890,10 @@ void RewardsServiceImpl::OnReconcileComplete(ledger::Result result,
   const std::string& viewing_id,
   ledger::REWARDS_CATEGORY category,
   const std::string& probi) {
+  if (contribution_count_ > 0) {
+    contribution_count_--;
+  }
+  BroadcastReconcile(contribution_count_ != 0);
```

Comment:
the convention is `TriggerEventName`, but I think you want to split this into two observer methods: `OnContributionStarted` and `OnContributionComplete`
`OnContributionInProgressChanged` isn't really correct because it's going to fire even when the status doesn't change (contribution_count_ is still != 0)

Reply:
which means you'd have `TriggerOnContributionStarted` and `TriggerOnContributionCompleted` which would call the corresponding observers

Reply:
Done.